### PR TITLE
LPS-200761 - Remove unwanted picklist entries during batch import

### PR DESCRIPTION
**Pull request by G-Sa**
Related to: [LPS-200761](https://liferay.atlassian.net/browse/LPS-200761)

### CODE REVIEW

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 1]
```diff
@@ -202,7 +204,7 @@ public static com.liferay.object.model.ObjectField toObjectField(
 		return serviceBuilderObjectField;
 	}
 
-	private static void _addListTypeEntries(
+	private static void _addOrUpdateListTypeEntries(
```
Comment (seltong) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:207]: My suggestion for the name of this method is `_addOrDeleteListTypeEntries`, what do you think?

**Review by seltong — CHANGES_REQUESTED** [review]

**Comment by seltong** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
Hey @carolmariaabb 
Could you review this PR, please? :smile:

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 2]
```diff
@@ -225,17 +227,21 @@ private static void _addListTypeEntries(
 			JSONArray objectStatesJSONArray = jsonObject.getJSONArray(
 				"objectStates");
 
+			List<String> serviceBuilderListTypeEntryKeys =
```
Comment (carolmariaabb) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:230]: @G-Sa rename it to `listTypeEntryKeys`

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java [hunk 3]
```diff
@@ -244,6 +250,11 @@ private static void _addListTypeEntries(
 					key,
 					Collections.singletonMap(LocaleUtil.getDefault(), key));
 			}
+
+			for (String key : serviceBuilderListTypeEntryKeys) {
```
Comment (carolmariaabb) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java:254]: @G-Sa rename it from `key` to `listTypeEntryKey`

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java [hunk 4]
```diff
@@ -107,6 +108,19 @@ public ListTypeEntry deleteListTypeEntry(long listTypeEntryId)
...
+		throws PortalException {
+
+		for (ListTypeEntry listTypeEntry :
+				listTypeEntryPersistence.findByListTypeDefinitionId(
+					listTypeDefinitionId)) {
+
+			if (StringUtil.equals(listTypeEntry.getKey(), key)) {
+				listTypeEntryLocalService.deleteListTypeEntry(listTypeEntry);
+			}
+		}
```
[the remark this answers is not in the document]
Comment (G-Sa) [reply modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java:121]: If i call the `listTypeEntryPersistence` to delete here, i'll also need to do the validation already done in the [original](https://github.com/liferay/liferay-portal/blob/master/modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java#L83-L97) `deleteListTypeEntry` method.

Other methods in this class also delegate that responsibility to the original method, wouldn't this break the pattern?

**Comment by carolmariaabb** [comment]
@G-Sa please see https://github.com/liferay-objects/liferay-portal/pull/2950

**Pull request closed without merging**